distributed/tests/test_scheduler.py

```diff
@@ -2442,3 +2443,23 @@ async def test_memory_no_zict(c, s, a, b):
 async def test_memory_no_workers(s):
     assert s.memory.process == 0
     assert s.memory.managed == 0
+
+
+@gen_cluster(client=True, nthreads=[])
+async def test_memory_is_none(c, s):
+    """If Worker.heartbeat() runs before Worker.monitor.update(), then
+    Worker.metrics["memory"] will be None and will need special handling in
+    Worker.memory and Scheduler.heartbeat_worker().
+    """
+    with mock.patch("distributed.system_monitor.SystemMonitor.update"):
+        w = await Worker(s.address, nthreads=1)
```

Comment:
Could you use a context manager here to ensure the `Worker` is shut down properly (or `await w.close()`)

Reply:
done